test/Bugs/misc_bugs.js

```diff
@@ -154,7 +154,17 @@ var tests = [
 
         var obj2 = {__proto__ : p}; // This should not call the getPrototypeOf
     }
-  }
+  },
+  {
+    name: "Destructuring declaration should return undefined",
+    body: function () {
+        assert.areEqual(undefined, eval("var {x} = {};"));
+        assert.areEqual(undefined, eval("let {x,y} = {};"));
+        assert.areEqual(undefined, eval("const [z] = [];"));
+        assert.areEqual(undefined, eval("let {x} = {}, y = 1, {z} = {};"));
+        assert.areEqual([1], eval("let {x} = {}; [x] = [1]"));

```

Comment:
This case is interesting to me. I guess returning 1 here wouldn't make sense if there were multiple assignments. Is a successful completion value then the array of all the assigned values?

Reply:
I think the last assignment's RHS will be assigned to the return register.

Reply:
Oh, I missed that this was two statements. That makes more sense. :)